napari/components/layerlist.py

```diff
@@ -257,21 +257,39 @@ def _step_size_from_scales(self, scales):
     def _get_step_size(self, layer_extent_list):
         if len(self) == 0:
             return np.ones(self.ndim)
-        else:
-            scales = [extent.step for extent in layer_extent_list]
-            min_scales = self._step_size_from_scales(scales)
-            return min_scales
 
-    @cached_property
-    def extent(self) -> Extent:
-        """Extent of layers in data and world coordinates."""
-        extent_list = [layer.extent for layer in self]
+        scales = [extent.step for extent in layer_extent_list]
+        return self._step_size_from_scales(scales)
+
+    def get_extent(self, layers: List[Layer]) -> Extent:
+        """
+        Return extent for a given layer list.
+        This function allows to calculate extent for subset of layers
+        when preparing and updating some technical layers.
+        For example Vectors cross in `multiple_viewer_widget` example
```

Comment:
```suggestion
        This function is useful for calculating the extent of a subset of layers
        when preparing and updating some supplementary layers.
        For example see the cross Vectors layer in the `multiple_viewer_widget` example.
```

I'm not sure what you mean by `technical` layer - maybe that some layers are not true data layers, but supplement those?